src/NestMemphis/nestMemphis.server.ts

```diff
@@ -0,0 +1,52 @@
+import { CustomTransportStrategy, Server } from '@nestjs/microservices';
+import { MemphisConnectionOptions } from './memphis.interface';
+import memphis, { MemphisType as Memphis, ConsumerType as Consumer } from '../memphis';
+
+export class MemphisServer extends Server implements CustomTransportStrategy {
+    private connection: Memphis;
+
+    private readonly options: MemphisConnectionOptions;
+
+    constructor(options: MemphisConnectionOptions) {
+        super();
+        this.options = options;
+    }
+
+    public async listen(callback: () => void): Promise<void> {
+        try {
+            this.connection = await memphis.connect(this.options);
```

Comment:
what is this? why are you creating a connection in here, we said only 1 connection is created for the entore application